controllers/csi-replicator/replicator_common.go

```diff
@@ -39,13 +39,16 @@ func shouldContinue(ctx context.Context, class *storageV1.StorageClass, driverNa
 		return false
 	}
 
-	// TODO: This should be removed when we start supporting RGs for SRDF Metro too
-	// Check for PowerMax SRDF Metro
-	if value, ok := class.Parameters["replication.storage.dell.com/RdfMode"]; ok {
-		if strings.ToUpper(value) == "METRO" {
-			log.V(common.InfoLevel).Info("Metro replication is not supported by Dell CSI Replication Controllers")
-			return false
-		}
+	// Check for PowerMax SRDF Metro and skip since it is only supported at the driver level
+	if value, ok := class.Parameters["replication.storage.dell.com/RdfMode"]; ok && strings.ToUpper(value) == "METRO" {
+		log.V(common.InfoLevel).Info("Metro replication is not supported by Dell CSI Replication Controllers")
+		return false
+	}
+
+	// Check for PowerStore Metro and skip since it is only supported at the driver level
+	if value, ok := class.Parameters["replication.storage.dell.com/mode"]; ok && strings.ToUpper(value) == "METRO" {
+		log.V(common.InfoLevel).Info("Metro replication is not supported by Dell CSI Replication Controllers")
+		return false
```

Comment:
We seem to be relying on assuming a minimum version of Replication for the driver. What could go wrong if they changed to use an older Replication version. What about pods changing during an upgrade/downgrade? Is there any danger of DU/DL?

Reply:
Driver and replicator side car runs in the same pod. Old pod with older versions will be servicing the request till the new one comes fully up and running. I don't think there will be any issue here.
It is recommended to use published versions for all drivers/modules in a CSM release. I cannot recollect the document location though.